pkg/cmd/codespace/list_test.go

```diff
@@ -1,15 +1,68 @@
 package codespace
 
 import (
+	"bytes"
 	"context"
 	"fmt"
 	"testing"
 
+	"github.com/cli/cli/v2/internal/browser"
 	"github.com/cli/cli/v2/internal/codespaces/api"
 	"github.com/cli/cli/v2/pkg/cmdutil"
 	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/google/shlex"
+	"github.com/stretchr/testify/assert"
 )
 
+func TestListCmdFlagError(t *testing.T) {
+	tests := []struct {
+		name     string
+		args     string
+		wantsErr error
+	}{
+		{
+			name:     "list codespaces,--repo, --org and --user flag",
+			args:     "--repo foo/bar --org github --user github",
+			wantsErr: fmt.Errorf("using `--org` or `--user` with `--repo` is not allowed"),
+		},
+		{
+			name:     "list codespaces,--web, --org and --user flag",
+			args:     "--web --org github --user github",
+			wantsErr: fmt.Errorf("using `--web` with `--org` or `--user` is not supported, please use with `--repo` instead"),
+		},
+		{
+			name:     "list codespaces, negative --limit flag",
+			args:     "--limit -1",
+			wantsErr: fmt.Errorf("invalid limit: -1"),
```

Comment:
Thanks for adding this missing test coverage!

Reply:
You're welcome.
Thank you for detailed review